src/plugins/s390.c

```diff
@@ -290,6 +290,62 @@ gboolean bd_s390_dasd_is_ldl (const gchar *dasd, GError **error) {
 }
 
 /**
+ * bd_s390_dasd_is_fba:
+ * @dasd: dasd to check, whether it is FBA
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a dasd is FBA
+ */
+gboolean bd_s390_dasd_is_fba (const gchar *dasd, GError **error) {
+    gchar *devname = NULL;
+    gint f = 0;
+    gint blksize = 0;
+    dasd_information2_t dasd_info;
+
+    memset(&dasd_info, 0, sizeof(dasd_info));
+
+    /* complete the device name */
+    if (g_str_has_prefix (dasd, "/dev/")) {
+        devname = g_strdup (dasd);
+    }
+    else {
+        devname = g_strdup_printf ("/dev/%s", dasd);
+    }
+
+    /* open the device */
+    if ((f = open(devname, O_RDONLY)) == -1) {
+        g_set_error (error, BD_S390_ERROR, BD_S390_ERROR_DEVICE,
+                     "Unable to open device %s", devname);
+        g_free (devname);
+        return FALSE;
+    }
+
+    g_free (devname);
+
+    /* check if this is a block device */
+    if (ioctl(f, BLKSSZGET, &blksize) != 0) {
+        close(f);
+        return FALSE;
+    }
+
+    /* get some info about DASD */
+    if (ioctl(f, BIODASDINFO2, &dasd_info) != 0) {
+        close(f);
+        return FALSE;
+    }
+
+    close(f);
+
+    /* check if we're on an FBA DASD */
+    if (strncmp(dasd_info.type, "FBA", 3) == 0) {
+        return TRUE;
+    }
+    else {
+        return FALSE;
```

Comment:
Please replace this ``if`` with ``return strncmp(dasd_info.type, "FBA", 3) == 0``.